Week1/assets/index.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+{
+  function fetchJSON(url, cb) {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url);
+    xhr.responseType = 'json';
+    xhr.onload = () => {
+      if (xhr.status < 400) {
+        cb(null, xhr.response);
+      } else {
+        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+      }
+    };
+    xhr.onerror = () => cb(new Error('Network request failed'));
+    xhr.send();
+  }
+
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach((key) => {
+      const value = options[key];
+      if (key === 'html') {
+        elem.innerHTML = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+  const repoLabels = ['Repository', 'Description', 'Forks', 'Updated'];
+  const repoValues = ['name', 'description', 'forks', 'updated_at'];
+  function renderRepo(repo)
+  {
+    const content = document.getElementById('content');
+    content.innerHTML = "";
+    const repoContent = createAndAppend('div', content, {class: 'item', id: 'repoInfo'});
+    const contributorsContent = createAndAppend('div', content, {class: 'item', id: 'contribInfo'});
+    createAndAppend('h2', contributorsContent, {html: 'Constributions'})
+
+    
+    repoLabels.forEach((element, index) => {
+
+      const item = createAndAppend('div', repoContent, {class:'item'});
+      createAndAppend('h3', item, {html: element +': '});
+      
+      const item1 = createAndAppend('div', repoContent, {class:'item'});
+     
+      if(element === 'Repository')
+        createAndAppend('a', item1, {html: repo[repoValues[index]], href: repo.url, target: '_blank'});
+      else if(element === 'Updated')
+      {
+        let myDate = new Date(repo[repoValues[index]]);
+        createAndAppend('h4', item1, {html: myDate.toLocaleString()});
```

Comment:
Here you have two main points I am referring to:
  1. `item` && `item1` is not a very described names so you can think about what they contain and what they exactly doing, read it as story :slightly_smiling_face:.
  2. Here is a very good example of what make people start confusing !!
   - let's just read it:
      - you have `let myDate // you have to choose a better name than myThing`
      - Which you are assigning to it the returning value of `new Date();`
      - Which you you are taking it from `repo` that t's parent object of `repoValues[index]`